<?xml version="1.0"?>
<launch>
  <arg name="bag_path" />

  <!-- Load robot description -->
  <param name="robot_description" command="$(find xacro)/xacro $(find asterix_description)/urdf/asterix_ugv.urdf.xacro" />
  <param name="robot_description_semantic" textfile="$(find asterix_moveit_config)/config/asterix_ugv.srdf" />

  <node pkg="hector_pose_prediction_benchmark" type="benchmark_node" name="pose_prediction_benchmark" output="screen" clear_params="true" ><!--launch-prefix="gdb -args"-->
    <param name="pose_predictor" value="sdf_contact_estimation"/>
    <param name="bag_file_path" value="$(arg bag_path)"/>
    <param name="result_folder" value="$(find hector_pose_prediction_benchmark)/data"/>
    <param name="path_sampling_resolution" value="0.05"/>
    <param name="wait_time" value="0.0"/>

    <!-- load map -->
    <rosparam subst_value="true" ns="sdf_contact_estimation/sdf_map">
<!--       scenario: step_0.18 -->
      voxel_size: 0.05
      use_esdf: true
      sdf_file_path: $(find sdf_contact_estimation)/sdf/gazebo_oc_pallet_terrain2.esdf
      truncation_distance: 0.4
    </rosparam>

    <!-- load robot model -->
    <rosparam file="$(find sdf_contact_estimation)/config/asterix_shape_config.yaml" command="load" ns="sdf_contact_estimation/shape_model" />

    <!-- load pose predictor -->
    <rosparam file="$(find sdf_contact_estimation)/config/optimization.yaml" command="load" ns="sdf_contact_estimation"/>
  </node>

  <!-- rviz ui -->
<!--   <include if="$(arg rviz)" file="$(find whole_body_posture_optimization)/launch/rviz.launch"/> -->

</launch>
